sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2792,6 +2792,51 @@ object GpuOverrides {
           aggBuffer.copy(dataType = c.dataType)(aggBuffer.exprId, aggBuffer.qualifier)
         }
       }),
+    expr[ApproximatePercentile](
+      "Approximate percentile",
+      ExprChecks.fullAggAndProject(
+        // note that output can be single number or array depending on whether percentiles param
+        // is a single number or an array
+        TypeSig.numeric + TypeSig.DECIMAL_64 +
+          TypeSig.ARRAY.nested(TypeSig.numeric + TypeSig.DECIMAL_64),
+        TypeSig.ARRAY.nested(TypeSig.all),
+        Seq(
+          ParamCheck("input", TypeSig.all, TypeSig.all),
+          ParamCheck("percentage",
+            TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 +
+              TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_64),
+            TypeSig.all),
+          ParamCheck("accuracy", TypeSig.commonCudfTypes + TypeSig.DECIMAL_64, TypeSig.all)
+        )),
+      (c, conf, p, r) => new TypedImperativeAggExprMeta[ApproximatePercentile](c, conf, p, r) {
+        override def tagExprForGpu(): Unit = {
+          if (childExprs.length < 2 || childExprs.length > 3) {
```

Comment:
This is uneeded. None of the child expressions are optional or a sequence so the length of childExprs will always be 3.  You can assert on that just to be sure if you want to.